File: daily_problems/2024/09/0925/personal_submission/cf1210b_yawn_sean.cpp
#include <bits/stdc++.h>
// #pragma GCC optimize("O3,Ofast,unroll-loops")
// #include <bits/extc++.h>
// using namespace __gnu_cxx;
// using namespace __gnu_pbds;
using namespace std;

signed main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    vector<long long> nums1(n);
    vector<int> nums2(n);
    map<long long, int> mp;

    for (auto &x: nums1) cin >> x, mp[x] ++;
    for (auto &x: nums2) cin >> x;

    vector<int> used(n, 0);
    long long ans = 0;

    for (auto &[k, v]: mp) {
        if (v > 1) {
            for (int i = 0; i < n; i ++) {
                if (used[i] == 0 && (nums1[i] & k) == nums1[i]) {
                    used[i] = 1;
                    ans += nums2[i];
                }
            }
        }
    }
    cout << ans;

    return 0;
}

File: daily_problems/2024/09/0925/personal_submission/cf1210b_yawn_sean.py
# Submission link: https://codeforces.com/contest/1210/submission/282845675
def main():
    n = II()
    nums1 = LII()
    nums2 = LII()

    cnt = Counter(nums1)
    used = [0] * n

    ans = 0
    for x in cnt:
        if cnt[x] > 1:
            for i in range(n):
                if not used[i] and nums1[i] & x == nums1[i]:
                    used[i] = 1
                    ans += nums2[i]

    print(ans)